File: NDPCommon/src/main/java/com/slender/validator/EmailValidator.java
package com.slender.validator;

import com.slender.annotation.Email;
import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

import java.util.regex.Pattern;

public class EmailValidator implements ConstraintValidator<Email, String> {
    @Override
    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
        if (email == null || email.isEmpty()) return true;
        return Pattern.compile("^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$").matcher(email).matches();
    }
}

File: NDPCommon/src/main/java/com/slender/validator/EmailValidator.kt
package com.slender.validator

import com.slender.annotation.Email
import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext
import java.util.regex.Pattern

class EmailValidator : ConstraintValidator<Email, String> {
    override fun isValid(
        email: String?,
        constraintValidatorContext: ConstraintValidatorContext?
    ): Boolean {
        if (email == null) return true
        return email.matches(Regex("^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$"))
    }
}
